pkg/clusterlogging/clusterlogging.go

```diff
@@ -0,0 +1,190 @@
+	Definition *clov1.ClusterLogging
+	// Created clusterLogging object on the cluster.
+	Object *clov1.ClusterLogging
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before clusterLogging object is created.
+	errorMsg string
+}
+
+// NewClusterLoggingBuilder method creates new instance of builder.
+func NewClusterLoggingBuilder(
+	apiClient *clients.Settings, name, nsname string) *ClusterLoggingBuilder {
+	glog.V(100).Infof("Initializing new clusterLogging structure with the following params: name: %s, namespace: %s",
+		name, nsname)
+
+	builder := &ClusterLoggingBuilder{
+		apiClient: apiClient,
+		Definition: &clov1.ClusterLogging{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the clusterLogging is empty")
+
+		builder.errorMsg = "The clusterLogging 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the clusterLogging is empty")
+
+		builder.errorMsg = "The clusterLogging 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// Pull returns clusterLogging object if found.
+func (builder *ClusterLoggingBuilder) Pull() (*clov1.ClusterLogging, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting clusterLogging %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	clusterLogging := &clov1.ClusterLogging{}
+	err := builder.apiClient.Get(context.Background(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, clusterLogging)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return clusterLogging, err
+}
+
+// Discover fetches existing clusterLogging from cluster.
+func (builder *ClusterLoggingBuilder) Discover() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Pulling existing clusterLogging with name %s under namespace %s from cluster",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	clo := &clov1.ClusterLogging{}
+	err := builder.apiClient.Get(context.Background(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, clo)
+
+	builder.Object = clo
+
+	return err
+}
+
+// Create makes a clusterLogging in the cluster and stores the created object in struct.
+func (builder *ClusterLoggingBuilder) Create() (*ClusterLoggingBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the clusterLogging %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
```

Comment:
Please add Get function to this package

Reply:
I don't wont to add methods I could not test, I add only verified code, I'm sorry.
Just because of the structure you decided each pkg should be build doesn't mean each pkg commit has to include all of it, this is the wrong approach. 
1. Code have to be tested before we commit it
2. not all of the methods, defined as must by your instructions, needed in case of each specific method implementation. Moreover, the implementation of some in specific areas makes the person who writes the code an idiot. Well, like delete clusterversion as an example (I could find more).
Not all such straight forward ...